### PR TITLE
docs: add jsdoc annotations everywhere

### PR DESCRIPTION
**Pull request by crisbeto**
Adds JSDoc annotations to all of the public properties and methods that were missing them. Also cleans up some random issues that were noticed while going through everything.

**Note:** I've skipped the toolbar component since @DevVersion mentioned that he's doing it already.

### CODE REVIEW

File: src/lib/button-toggle/button-toggle.ts [hunk 1]
```diff
@@ -98,10 +98,12 @@ export class MdButtonToggleGroup implements AfterViewInit, ControlValueAccessor
   @ContentChildren(forwardRef(() => MdButtonToggle))
   _buttonToggles: QueryList<MdButtonToggle> = null;
 
+  /** @docs-private */
   ngAfterViewInit() {
```
Comment (DevVersion) [inline src/lib/button-toggle/button-toggle.ts]: All the lifecycle hooks should be already filtered in [this processor](https://github.com/angular/material2/blob/master/tools/dgeni/processors/docs-private-filter.js#L6-L14)

File: src/lib/core/a11y/interactivity-checker.ts [hunk 2]
```diff
@@ -15,8 +16,13 @@ export class InteractivityChecker {
 
   constructor(private _platform: Platform) {}
 
-  /** Gets whether an element is disabled. */
-  isDisabled(element: HTMLElement) {
+  /**
+   * Gets whether an element is disabled.
+   *
+   * @param element Element to be checked.
+   * @returns {boolean} Whether the element is disabled.
```
Comment (DevVersion) [inline src/lib/core/a11y/interactivity-checker.ts]: We don't need the type here

File: src/lib/core/a11y/interactivity-checker.ts [hunk 3]
```diff
@@ -27,16 +33,21 @@ export class InteractivityChecker {
    *
    * This will capture states like `display: none` and `visibility: hidden`, but not things like
    * being clipped by an `overflow: hidden` parent or being outside the viewport.
+   *
+   * @returns {boolean} Whether the element is visible.
```
Comment (DevVersion) [inline src/lib/core/a11y/interactivity-checker.ts]: Type is unnecessary in TS (same for below and everywhere else)

File: src/lib/core/observe-content/observe-content.ts [hunk 4]
```diff
@@ -9,16 +9,22 @@ import {
   AfterContentInit
 } from '@angular/core';
 
+/**
+ * Directive that triggers a callback whenever the content of
+ * it's associated element has changed.
```
Comment (DevVersion) [inline src/lib/core/observe-content/observe-content.ts]: Should be `its`?

File: src/lib/core/a11y/interactivity-checker.ts (hunk 2 above)
Comment (crisbeto) [inline src/lib/core/a11y/interactivity-checker.ts]: We do, dgeni doesn't figure out the return types, it was requested by @jelbourn.

File: src/lib/core/a11y/interactivity-checker.ts (hunk 3 above)
Comment (crisbeto) [inline src/lib/core/a11y/interactivity-checker.ts]: We do since dgeni doesn't figure out the types.

**Comment by jelbourn** [comment]
LGTM; I'll make any tweaks after merging

**Pull request merged**